packages/vite/src/node/plugins/css.ts

```diff
@@ -464,15 +464,13 @@ export function cssPostPlugin(config: ResolvedConfig): Plugin {
       let isPureCssChunk = true
       const ids = Object.keys(chunk.modules)
       for (const id of ids) {
-        if (
-          !isCSSRequest(id) ||
-          cssModuleRE.test(id) ||
-          commonjsProxyRE.test(id)
-        ) {
+        if (cssModuleRE.test(id)) {
           isPureCssChunk = false
         }
         if (styles.has(id)) {
           chunkCSS += styles.get(id)
+        } else {
+          isPureCssChunk = false
```

Comment:
We can omit the `isCssRequest` and `commonjsProxyRE` checks because `styles` will not be populated if they match, acting as a pseudo-cache.

However `cssModuleRE` is still checked because it adds to `styles` but causes the chunk to not be a pure css chunk

Reply:
Would you add a comment about this on line 473? I think it could be hard for others to follow the code if not.

Reply:
Added 👍 I also move the `cssModuleRE` check inside `styles.has(id)` which makes more sense and easier to understand why.